README.md
# flutter_time_ago

A new Flutter package to format date or milliseconds in seconds ago, minutes ago, hours ago, days ago and from now.

## Usage

Format any `DateTime` or `milliseconds` format by following steps:

```dart
import 'package:flutter_time_ago/flutter_time_ago.dart';

var _dateTime = DateTime.now().subtract(Duration(seconds: 300));
print(FlutterTimeAgo.parse(_dateTime)); // 5 minutes ago
print(FlutterTimeAgo.parse(1605985200000, lang:'pt_BR')); // há 1 mês
```

### Formatting from fromMilliseconds

if you have a date in milliseconds and the type is String, you need to convert it to int ex: int.parse('1605985200000'):

```dart
import 'package:flutter_time_ago/flutter_time_ago.dart';

int milliseconds = int.parse('1605985200000');
print(FlutterTimeAgo.parse(milliseconds)); // 1 month ago
```

### Setting default locale

If you want to change default `lang`, then call `setConfig` method into a function.\
Languages available so far: pt_BR', 'pt_BR_short', 'en', 'en_short', 'es', 'es_short'

```dart
import 'package:flutter_time_ago/flutter_time_ago.dart';

@override
  void initState() {
    super.initState();
     FlutterTimeAgo.setConfig(
     'pt_BR',
     /// Remove a prefix text ex: [in] 2 minutes
     /// Default is [true]
     prefixEnabled:false,
     /// Remove a suffix text ex: 2 minutes [ago] or [from now]
     /// Default is [true]
     suffixEnabled:false,
     );
  }
```

### Your language is not supported?

if your language is not supported, you can create your own, just create a class that implements LocaleMessagesInterface

```dart
import 'package:flutter_time_ago/flutter_time_ago.dart';

class CustomLocale implements LocaleMessagesInterface {...}
```

### how to you set custom language?

```dart
@override
  void initState() {
    super.initState();
     FlutterTimeAgo.setCustomLocale(
        {'fr': CustomLocale(), 'fr_short': CustomShortLocale()}
        );
  }
```

### Using FlutterTimeAgoWidget

A widget that rebuilds itself every 1 minute in periods to update the time on the screen

```dart
@override
 Widget build(BuildContext context) {
 return FlutterTimeAgoWidget(
        date: 1605985200000,
        /// Remove a prefix text ex: [in] 2 minutes
        /// Default is [true]
        prefixEnabled:false,
        /// Remove a suffix text ex: 2 minutes [ago] or [from now]
        /// Default is [true]
        suffixEnabled:false,
        textAlign: TextAlign.center,
        textStyle: TextStyle(
        color: Colors.grey,
        fontWeight: FontWeight.w600,
          ),
        );
 }
```

don't forget to look at our example application in the /example folder\
just run the `flutter run` command inside it


CHANGELOG.md
## [0.0.1] - Initial release.

- Initial release.

## [0.0.2] - Initial release.

- Initial release.

## [0.0.3] - Add FlutterTimeAgoWidget.

- A widget that rebuilds itself every 1 minute in periods to update the time on the screen.


example/lib/main.dart
import 'package:example/custom_locale.dart';
import 'package:flutter/material.dart';
import 'package:flutter_time_ago/flutter_time_ago.dart';

import 'page/test_page.dart';

void main() {
  FlutterTimeAgo.setConfig('pt_BR');
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'TimeAgo Example',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: HomeView(),
    );
  }
}

class HomeView extends StatefulWidget {
  @override
  _HomeViewState createState() => _HomeViewState();
}

class _HomeViewState extends State<HomeView> {
  String selectedLang = 'pt_BR';
  List<dynamic> times;
  List<String> localesList = [
    'pt_BR',
    'pt_BR_short',
    'en',
    'en_short',
    'es',
    'es_short',
    'fr',
    'fr_short',
  ];

  @override
  void initState() {
    DateTime now = DateTime.now();
    times = [
      now,
      now.subtract(
        Duration(minutes: 25),
      ),
      now.subtract(
        Duration(hours: 3),
      ),
      now.subtract(
        Duration(days: 2),
      ),
      now.subtract(
        Duration(days: 8),
      ),
      now.subtract(
        Duration(days: 40),
      ),
      now.subtract(
        Duration(days: 365),
      ),
      now,
      now.add(
        Duration(minutes: 25),
      ),
      now.add(
        Duration(hours: 3),
      ),
      now.add(
        Duration(days: 2),
      ),
      now.add(
        Duration(days: 8),
      ),
      now.add(
        Duration(days: 40),
      ),
      now.add(
        Duration(days: 365),
      ),
      1609302107000,
      1609298507000,
      1608034642938,
      1605985200000
    ];
    FlutterTimeAgo.setCustomLocale(
        {'fr': CustomLocale(), 'fr_short': CustomShortLocale()});
    super.initState();
  }

  List<dynamic> get _timesToString => times
      .map((time) => time.runtimeType == DateTime ? time.toString() : time)
      .toList();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Container(
                height: 100,
                width: double.infinity,
                alignment: Alignment.center,
                color: Colors.grey[400],
                child: ListView(
                  scrollDirection: Axis.horizontal,
                  children: localesList
                      .map(
                        (e) => Padding(
                          padding: const EdgeInsets.all(5),
                          child: ActionChip(
                            onPressed: () {
                              FlutterTimeAgo.setConfig(
                                e,
                                prefixEnabled: false,
                                suffixEnabled: true,
                              );
                              setState(() {
                                selectedLang = e;
                              });
                            },
                            backgroundColor: identical(e, selectedLang)
                                ? Colors.blue
                                : Colors.grey[300],
                            labelStyle: TextStyle(
                              color: identical(e, selectedLang)
                                  ? Colors.white
                                  : Colors.black,
                              fontWeight: FontWeight.bold,
                            ),
                            label: Text(e),
                          ),
                        ),
                      )
                      .toList(),
                ),
              ),
              Padding(
                padding:
                    const EdgeInsets.symmetric(vertical: 15, horizontal: 8),
                child: Text(
                  'EXEMPLOS',
                  style: TextStyle(
                    color: Colors.grey,
                    fontWeight: FontWeight.w600,
                    letterSpacing: 2,
                  ),
                ),
              ),
              ListView.builder(
                shrinkWrap: true,
                itemCount: _timesToString.length,
                padding: EdgeInsets.zero,
                physics: NeverScrollableScrollPhysics(),
                itemBuilder: (BuildContext context, int index) => Center(
                  child: Padding(
                    padding: const EdgeInsets.all(2),
                    child: FlutterTimeAgoWidget(
                      date: _timesToString[index],
                      textAlign: TextAlign.center,
                      textStyle: TextStyle(
                        color: Colors.grey,
                        fontWeight: FontWeight.w600,
                      ),
                    ),
                  ),
                ),
              ),
              OutlineButton(
                onPressed: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(
                        builder: (context) => TestPage(
                              items: _timesToString,
                            )),
                  );
                },
                child: Text('Go Page Now'),
              )
            ],
          ),
        ),
      ),
    );
  }
}
